test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -12639,17 +12639,18 @@ def isUp(self):
         """
         Check for PBS mom up
         """
-        # Poll for few seconds so we don't miss any signals sent to
-        # mom
-        for _ in range(5):
+        # Poll for few seconds to see if mom is up and node is free
+        for _ in range(20):
             rv = super(MoM, self)._isUp(self)
             if rv:
                 try:
                     nodes = self.server.status(NODE, id=self.shortname)
                     if nodes:
                         self.server.expect(NODE, {'state': 'free'},
```

Comment:
looking at it again I think you should keep the check for node state=free outside of the loop.

wait till mom is up.
if mom is up then wait till node state is free.

Reply:
@anamikau , I have made the change. Please take a look